src/transformers/generation/logits_process.py

```diff
@@ -1085,20 +1085,134 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
+    - **Penalizing Repeated Tokens**: If a beam in a group selects a token already chosen by another group in the same
+      step, a penalty is applied to that token's score.
+    - **Promoting Diversity**: This penalty discourages beams within a group from selecting the same tokens as beams in
+      other groups.
+
+    Benefits:
+    - **Diverse Outputs**: Produces a variety of different sequences.
+    - **Exploration**: Allows the model to explore different paths.
 
     Args:
         diversity_penalty (`float`):
             This value is subtracted from a beam's score if it generates a token same as any beam from other group at a
             particular time. Note that `diversity_penalty` is only effective if `group beam search` is enabled.
+                    -- The penalty applied to a beam's score when it generates a token that has already been chosen
+                            by another beam within the same group during the same time step.
+                    -- A higher `diversity_penalty` will enforce greater diversity among the beams,
+                            making it less likely for multiple beams to choose the same token.
+                    -- Conversely, a lower penalty will allow beams to more freely choose similar tokens. -- Adjusting
+                    this value can help strike a balance between diversity and natural likelihood.
         num_beams (`int`):
             Number of beams used for group beam search. See [this paper](https://arxiv.org/pdf/1610.02424.pdf) for more
             details.
+                    -- Beam search is a method used that maintains beams (or "multiple hypotheses") at each step,
+                            expanding each one and keeping the top-scoring sequences.
+                    -- A higher `num_beams` will explore more potential sequences -- This can increase chances of
+                    finding a high-quality output but also increases computational cost.
         num_beam_groups (`int`):
             Number of groups to divide `num_beams` into in order to ensure diversity among different groups of beams.
             See [this paper](https://arxiv.org/pdf/1610.02424.pdf) for more details.
+                    -- Each group of beams will operate independently, selecting tokens without considering the choices
+                    of other groups. -- This division promotes diversity by ensuring that beams within different groups
+                    explore different paths. -- For instance, if `num_beams` is 6 and `num_beam_groups` is 2, there
+                    will be 2 groups each containing 3 beams. -- The choice of `num_beam_groups` should be made
+                    considering the desired level of output diversity and the total number of beams.
+
+
+    Example: the below example shows a comparison before and after applying Hamming Diversity.
+
+    ```python
+            from transformers import AutoTokenizer, AutoModelForSeq2SeqLM
```

Comment:
Final nit 1: let's remove indentation in the example, it shows up awkwardly in the rendered docs (see [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_25481/en/internal/generation_utils#transformers.HammingDiversityLogitsProcessor) the preview)

Reply:
oh, yeah, and the missing `>>>` / `...`, good catch!